scripts/courseGenerator/CourseGeneratorInterface.lua

```diff
@@ -204,12 +218,10 @@ function CourseGeneratorInterface.generateDefaultCourse(nHeadlands)
     settings.workWidth:refresh()
     settings.workWidth:setFloatValue(width)
     vehicle:getCpSettings().toolOffsetX:setFloatValue(offset)
-    if nHeadlands then
-        settings.numberOfHeadlands:setFloatValue(nHeadlands)
-    end
     settings.sharpenCorners:setValue(true)
     CpUtil.infoVehicle(vehicle, "Generating default course with %d headlands", settings.numberOfHeadlands:getValue())
-    local ok, course = CourseGeneratorInterface.generate(points, {x = x, z = z}, vehicle, settings)
+    local x, _, z = getWorldTranslation(vehicle.rootNode)
+    local ok, course = CourseGeneratorInterface.generate(fieldPolygon, {x = x, z = z}, vehicle, settings, islandPolygons)
```

Comment:
The start position should probably also be passed as argument instead of the vehicle position?